multiqc/modules/picard/HsMetrics.py

```diff
@@ -108,7 +148,87 @@ def parse_reports(self):
+                'content': table_html})
+        self.sections.append(_add_target_bases(data))
+        self.sections.append(_add_hs_penalty(data))
+        # Return the number of detected samples to the parent module
     return len(self.picard_HsMetrics_data)
-    
\ No newline at end of file
+
+def _clean_table(data):
+    data_table = defaultdict(dict)
+    for s in data:
+        for h in data[s]:
+            if h.startswith("PCT"):
+                continue
+            if h.startswith("HS"):
+                continue
+            if h == "GENOME_SIZE":
+                continue
+            data_table[s][h] = data[s][h]
+    return data_table
+
+def _get_headers(data):
+    header = {}
+    for s in data:
+        for h in data[s]:
+            try:
+                float(data[s][h])
+            except:
+                continue
+            if h not in header:
+                this = {
+                'title': h.replace("_", " ").lower().capitalize(),
+                'description' : DESC[h] if h in DESC else "",
+                'scale': 'RdYlGn'
+                }
+                if h.startswith("PCT"):
+                    this.update({'max': 100, 'min': 0, 'format': '{:.0f}%', 'scale': 'RdYlGn'})
+                elif h.find("READS") > -1:
+                    this.update({'shared_key': "read_count",
+                                 'modify': lambda x: x / 1000000.0})
+                    this['title'] = "M {0}".format(this['title'])
+                elif h.find("BASES") > -1:
+                    this.update({'shared_key': 'bases_count',
+                                 'modify': lambda x: x / 1000000.0})
+                    this['title'] = "Mb {0}".format(this['title'])
+                if h in ["BAIT_TERRITORY", "TOTAL_READS", "TARGET_TERRITORY", "AT_DROPOUT", "GC_DROPOUT"]:
+                    this.update({'hidden': True})
+                header.update({h : this})
+    return OrderedDict(sorted(header.items(), key=lambda t: t[1]['title']))
+
+def _add_target_bases(data):
+    subtitle = "<p>The percentage of all target bases achieving `NUMBER`X or greater coverage.</p>"
+    data_clean = defaultdict(dict)
+    for s in data:
+        for h in data[s]:
+            if h.startswith("PCT_TARGET"):
+                data_clean[s][int(h.replace("PCT_TARGET_BASES_", "")[:-1])] = data[s][h] * 100.0
+    pconfig = { 'xlab': 'Number of reads',
+                'ylab': 'Pct of bases'}
+    return {'name': 'Target Region Coverage',
+    'anchor': 'picard_hsmetrics_target_bases',
+    'content': subtitle + plots.linegraph.plot(data_clean, pconfig)}
+
+def _add_hs_penalty(data):
+    subtitle = "<p>The hybrid selection penalty incurred to get 80% of target bases to 10X. This metric should be interpreted as: if I have a design with 10 megabases of target, and want to get 10X coverage I need to sequence until PF_ALIGNED_BASES = 10^7 * 10 * HS_PENALTY_`NUMBER`X.</p>"
+    data_clean = defaultdict(dict)
+    any_non_zero = False
```

Comment:
Is this used? Would be nice - I have all zeroes in my demo data but I get the plot still.